src/runtime/GlobalObjectBuiltinRegExp.cpp

```diff
@@ -350,6 +350,59 @@ static Value builtinRegExpSplit(ExecutionState& state, Value thisValue, size_t a
     return A;
 }
 
+#ifdef ESCARGOT_ENABLE_ES2015
+static Value builtinRegExpMatch(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    RESOLVE_THIS_BINDING_TO_OBJECT(thisObject, RegExp, match);
+
+    String* str = argv[0].toString(state);
+    ASSERT(str != nullptr);
+    RegExpObject* rx;
+
+    if (thisObject->isRegExpObject(state) == true) {
+        rx = thisObject->asRegExpObject(state);
+    } else {
+        rx = new RegExpObject(state, Value(thisObject).toString(state), String::emptyString);
+    }
+    //21.2.5.6.8
+    bool global = rx->option() & RegExpObject::Option::Global;
+    if (global == false) {
+        return builtinRegExpExec(state, rx, 1, &argv[0], true);
+    } else {
+        bool fullUnicode = rx->option() & RegExpObject::Option::Unicode;
+        rx->setLastIndex(state, Value(0));
```

Comment:
At this point the this call only can be used, if `rx` is a RegExp object, however, this routine is generic, so you should handle the non-RegExp objects as well. So please use the generic `[[Set]]` method here.